CourseOne/main.swift

```diff
@@ -7,5 +7,238 @@
 
 import Foundation
 
-print("Hello, World!")
+// 2) Повторить то, что есть в видео.
+/*
+ Замыкания являются анонимными функциями, которые можно передавать в качестве аргументов другим функциям
+ {
+    (парамеиры) -> возврвщвемый_тип in
+        операторы
+ }
+ */
 
+var names = ["Максим", "Сергей", "Николай", "Артем", "альберт", "Том"]
+let s = names.sorted()
+
+// (String, String) -> Bool
+
+let t1 = names.sorted { (s1: String, s2: String) -> Bool in
+    s1 < s2
+}
+// Сокращаем запись
+let t2 = names.sorted {  (s1, s2) -> Bool in
+    s1 < s2
+}
+// Сокращаем запись
+let t3 = names.sorted { s1, s2 in s1 < s2 }
+
+// Сортировка в обратном порядке
+let t4 = names.sorted { s1, s2 in s2 < s1 }
+
+// Сортировка по длине символов
+let t5 = names.sorted { s1, s2 in s1.count < s2.count }
+print(t5)
+
+// Автоматические имена аргументов
+let p1 = names.sorted { $0 < $1 }
+print(p1)
+
+// Захват значений
+func makeTranslator(stroka: String) -> (String) -> (String) {
+    
+    return { (name: String) -> String in return (stroka + " " + name) }
+}
+
+var englishWelcom = makeTranslator(stroka: "Привет")
+print(englishWelcom("Jack"))
+
+//
+var wallet = [10, 50, 100, 100, 5000, 100, 50, 50, 500, 100]
+// проверка массива на содержание купюры номинали 100 и вывести в отделный массив
+func handle100(_ wallet: [Int]) -> [Int] {
+    var returnWallet: [Int] = []
+    for banknote in wallet {
```

Comment:
```suggestion
    for banknote in wallet where banknote == 100 {
```
я бы попробовал вот так сделать. В этом контексте удобнее условие написать, что бы не было лишних вложенностей.